src/main/java/org/minions/devfund/damianvp/moviedatabase/Actor.java

```diff
@@ -0,0 +1,91 @@
+package org.minions.devfund.damianvp.moviedatabase;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Actor class.
+ */
+public class Actor {
+    private String name;
+    private List<Movie> movies;
+
+    /**
+     * Class Constructor.
+     * @param name String type.
+     */
+    Actor(final String name) {
+        this.name = name;
+        this.movies = new ArrayList<>();
+    }
+
+    /**
+     * Class Constructor, empty values.
+     */
+    Actor() {
+        this.name = "";
+        this.movies = new ArrayList<>();
+    }
+
+    /**
+     * Class Constructor.
+     * @param name  String type.
+     * @param movies  ArrayList of Movie Objects.
+     */
+    Actor(final String name, final ArrayList<Movie> movies) {
+        this.name = name;
+        this.movies = movies;
+    }
+
+    /**
+     * Method to get the Actor name.
+     * @return String, name property.
+     */
+    String getName() {
+        return this.name;
+    }
+
+    /**
+     * Method to set the name.
+     * @param name String type.
+     */
+    void setName(final String name) {
+        this.name = name;
+    }
+
+    /**
+     * Method to return the Rating Average of current Actor.
+     * @return Double, average of Movie ratings.
+     */
+    Double getRatingAverage() {
+        return this.movies.stream().mapToDouble(Movie::getRating).average().orElse(0.0);
+    }
+
+    /**
+     * Method to get the Movies.
+     * @return ArrayList<Movie>
+     */
+    List<Movie> getMovies() {
+        return this.movies;
+    }
+
+    /**
+     * Method to add a movie in arrayList<Movie>.
+     * @param movie Movie object.
+     */
+    void addMovie(final Movie movie) {
+        if (!movieExists(movie.getName())) {
```

Comment:
This validation is performed twice 
the first time on MovieDatabase.addMovie
Please remove it in order to increase the code coverage

Reply:
I have some doubts with this, yes there are 2 validations but every is for different class, and for example, currently I have specifically a unit test where is tested the adding repeated movies, it is  calling directly from actor class if I deleted it, that test fails